### PR TITLE
Allow reading HFSS files from StringIO

### PR DESCRIPTION
**Pull request by FranzForstmayr**
This patch fixes reading hfss files from stringio.

See #456 

The `get_gamma_z0_fid` method is now called before closing the stringIO handle, the values are stored in the Touchstone class.
The original `get_gamma_z0` returns just the previously parsed values.

### CODE REVIEW

File: skrf/io/touchstone.py [hunk 1]
```diff
@@ -459,9 +468,9 @@ def is_from_hfss(self):
...
         return False
 
-    def get_gamma_z0(self):
+    def get_gamma_z0_from_fid(self, fid):
         '''
-        Extracts Z0 and Gamma comments from touchstone file (is provided)
+        Extracts Z0 and Gamma comments from fid
         
         Returns
         --------
```
Comment (jhillairet) [inline skrf/io/touchstone.py:469]: if I'm not wrong, this function doesn't return anything anymore

**Review by jhillairet — COMMENTED** [review]
Nice ! Indeed, the previous change to make Network() compatible with Stream forgot this part.

I'm curious : in what kind of application did you need to read HFSS files from streams ?

File: skrf/tests/test_network.py [hunk 2]
```diff
@@ -141,7 +141,14 @@ def test_constructor_from_stringio(self):
         with open(filename) as fid:
             data = fid.read()
             sio = io.StringIO(data)
-            sio.name = filename # hack a bug to touchstone reader
+            sio.name = os.path.basename(filename) # hack a bug to touchstone reader
+            rf.Network(sio)
+            
+        filename = os.path.join(self.test_dir, 'hfss_oneport.s1p')
```
[the remark this answers is not in the document]
Comment (FranzForstmayr) [reply skrf/tests/test_network.py:147]: I can add some more testcases!

File: skrf/io/touchstone.py [hunk 3]
```diff
@@ -105,8 +105,24 @@ def __init__(self, file):
...
+
+        self.gamma = []
+        self.z0 = []
+
+        if self.is_from_hfss:
+            self.get_gamma_z0_from_fid(fid)
+
         fid.close()
 
+    def get_line(self, fid):
```
[the remark this answers is not in the document]
Comment (FranzForstmayr) [reply skrf/io/touchstone.py]: I removed the `get_line` method again, as it's simply a for loop. The check for "python3 zipfile compatibility" is not necessary anymore. (All tests pass obviously)

**Comment by FranzForstmayr** [comment]
We have an web (Dash & Plotly) touchstone viewer in our company, the files are uploaded to a server and converted to streams to avoid temporary files.
Unfortunately I'm not allowed to make the viewer public, however I need lot's of stuff from skrf and I'm happy if I can help here :).

**Comment by FranzForstmayr** [comment]
And beside measured sNp file, we have lots of hfss generated files, which should not throw errors :)

**Comment by jhillairet** [comment]
> We have an web (Dash & Plotly) touchstone viewer in our company, the files are uploaded to a server and converted to streams to avoid temporary files.
> Unfortunately I'm not allowed to make the viewer public, however I need lot's of stuff from skrf and I'm happy if I can help here :).

Nice. Have a look at https://github.com/scikit-rf/dash-apps if you haven't already

File: skrf/io/touchstone.py [hunk 4]
```diff
@@ -459,68 +461,82 @@ def is_from_hfss(self):
 
         return False
 
-    def get_gamma_z0(self):
+    def get_gamma_z0_from_fid(self, fid):
         '''
-        Extracts Z0 and Gamma comments from touchstone file (is provided)
+        Extracts Z0 and Gamma comments from fid
```
Comment (arsenovic) [inline skrf/io/touchstone.py:466]: should we attempt some backward compatability here?

Comment (FranzForstmayr) [reply skrf/io/touchstone.py:466]: I didn't change `get_gamma_z0_from_fid`, this method is new. The public method `get_gamma_z0` has not changed the interface, it returns exactly the same as before.

Comment (arsenovic) [reply skrf/io/touchstone.py:466]: ok great . thanks.

**Comment by jhillairet** [comment]
@FranzForstmayr OK for merging?

**Comment by jhillairet** [comment]
The changes led to break compatibility with `NetworkSet.from_zip()` method. Interestingly, the problem was not visible until we merged the PR, as this method was not tested and used in only one notebook. Since the notebook testing with nbsmoke was performed using the master branch version of skrf, it appeared OK...

I'll make new PRs to improve the notebook testing, to fix NetworkSet and add tests for NetworkSet as I realized that no test exist at all for it.

**Pull request merged**